### PR TITLE
Add spike counts to pre-commit tests.

### PR DESCRIPTION
**Pull request by thorstenhater**
- Enhance `run_cpp_examples.sh` to do some very basic checking of spikes counts
- Add a script to check all releases
- The former must succeed for all commits to master

Here's the expected spike counts as of 0.7
- brunel 6998
- bench 972
- ring 94 (19 pre-v0.7)
- gap_junctions 30

### CODE REVIEW

File: scripts/check-all-tags.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if [ "$expected" == "$actual" ]
+    then
+        echo "   - $prog: OK"
+    else
+        echo "   - $prog: ERROR wrong number of spikes: $expected ./. $actual"
+        ok=1
+    fi
+}
+
+for tag in "v0.4-79855b66" "v0.5.2-51e35898" "v0.6-930c23eb" "v0.7-d0e424b4"
```
Comment (brenthuisman) [inline scripts/check-all-tags.sh:57]: This is not just the tag but a tag-hash tuple. Is that necessary? Loop above seems to do well w/o.

File: doc/contrib/release.rst [hunk 2]
```diff
@@ -75,6 +75,8 @@ Release
    
 #. Change ``VERSION``. Make sure does not end with ``-rc`` or ``-dev``.
 
+#. Update ``scripts/check-all-tags.sh`` to check the current tag.
```
Comment (brenthuisman) [inline doc/contrib/release.rst:78]: Automated testing will have caught any changes by now, so all that's left is to add nextver to the loops, right?

**Review by brenthuisman — CHANGES_REQUESTED** [review]
Noice.

File: doc/contrib/release.rst (hunk 2 above)
Comment (thorstenhater) [reply doc/contrib/release.rst:78]: Yes, just add `tag-hash`. The check-all script is for our manual regression testing, it's not used in an automated workflow.

File: scripts/check-all-tags.sh (hunk 1 above)
Comment (thorstenhater) [reply scripts/check-all-tags.sh:57]: The tags are just for human convenience and/or double checking.

**Review by brenthuisman — APPROVED** [review]

**Pull request merged**